hms-dbmi-cellenics/ui
# [BIOMAGE-1684] - Add DE warning

# Description
#### Background
Running between-group DE comparison requires at least 2 samples to run. For comprehensive statistitics, we need the cells involved in the comparison to come from at least 3 different samples.

This PR provides a warning to the user to let them know if the group they have chosen to compare does not contain enough cells for complete statistical analysis.

#### To test
To test the check that is implemented here, you need to have a project with at least 4 samples.

- Go to Data Exploration, open the Differential Expression tab.
- Select comparison. between groups
- Choose a cluster located somewhere in the early middle of the cluster list (Cluster 0).
- Select one of the samples for the first group to compare
- Select "all" or "rest of samples".
- Change the cluster that is used as the basis.  You might be able to trigger the warning and the error alert.

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-1684

#### Link to staging deployment URL (or set N/A)
https://ui-agi-650.scp-staging.biomage.net

#### Links to any PRs or resources related to this PR

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR

### Manual/unit testing


### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.


### Documentation updates

### Optional
